pallets/eth-bridge/src/eth.rs

```diff
@@ -176,33 +185,33 @@ fn to_param_type(key: &Vec<u8>) -> Option<ParamType> {
     }
 }
 
-fn to_token_type<T: pallet::Config>(kind: &ParamType, value: &Vec<u8>) -> Result<Token, Error<T>> {
+fn to_token_type<T: pallet::Config>(kind: &ParamType, value: &[u8]) -> Result<Token, Error<T>> {
     match kind {
         ParamType::Uint(_) => {
-            let dec_value = Int::from_dec_str(&String::from_utf8(value.clone()).unwrap())
-                .map_err(|_| Error::<T>::InvalidUint)?;
+            let dec_str = core::str::from_utf8(value).map_err(|_| Error::<T>::InvalidUtf8)?;
+            let dec_value = Int::from_dec_str(dec_str).map_err(|_| Error::<T>::InvalidUint)?;
```

Comment:
It did. All these types were covered in the early tests and they ran just fine then. The tests have been commented out since and need fixing and reinstating to be certain though.

Reply:
Update: fixed tests are now included in this PR